src/pybind/mgr/cephadm/module.py

```diff
@@ -2859,6 +2861,36 @@ def set_prometheus_access_info(self, user: str, password: str) -> str:
         self.set_store(PrometheusService.USER_CFG_KEY, user)
         self.set_store(PrometheusService.PASS_CFG_KEY, password)
         return 'prometheus credentials updated correctly'
+    
+    @handle_orch_error
+    def set_prometheus_target(self, url: str) -> str:
+        prometheus_spec = cast(PrometheusSpec, self.spec_store['prometheus'].spec)
+        if url not in prometheus_spec.targets:
+            prometheus_spec.targets.append(url)
+        else:
+            raise OrchestratorError('Target {} already exists'.format(url))
+        if not prometheus_spec:
+            raise OrchestratorError('Service prometheus not found')
+        daemons: List[orchestrator.DaemonDescription] = self.cache.get_daemons_by_type('prometheus')
+        spec = ServiceSpec.from_json(prometheus_spec.to_json())
+        self.apply([spec], no_overwrite=False)
+        self.daemon_action(action='redeploy', daemon_name=daemons[0].daemon_name)
```

Comment:
why only redeploy the first prometheus daemon, rather than all, in the case where there are multiple (same for remove-target)?

Reply:
Makes sense, updated the PR